packages/react/src/utils/useCustomPages.tsx

```diff
@@ -54,13 +58,18 @@ const useCustomPages = ({
   children,
   LinkComponent,
   PageComponent,
+  MenuItemsComponent,
   reorderItemsLabels,
   componentName,
 }: UseCustomPagesParams) => {
   const validChildren: CustomPageWithIdType[] = [];
 
   React.Children.forEach(children, child => {
-    if (!isThatComponent(child, PageComponent) && !isThatComponent(child, LinkComponent)) {
+    if (
+      !isThatComponent(child, PageComponent) &&
+      !isThatComponent(child, LinkComponent) &&
+      !isThatComponent(child, MenuItemsComponent)
+    ) {
```

Comment:
❓ Do we allow for  `<UserProfile.MenuItems/>` ? What's the reason we need this ?

Reply:
@panteliselef We can define custom pages through `UserButton` component and we need to allow `MenuItems` as children component. 
For custom pages we use the `useUserProfileCustomPages` for both `UserProfile` and `UserButton`, maybe we should create 2 separate utils here in order to be able to determine the correct component and validate the component only if is it `UserButton`, wdyt?

Reply:
After our discussion, I confirm that this change is necessary. Regarding your question, this is definitely a good suggestion but nothing that needs to delay this PR any longer.